### PR TITLE
Add GCP IoT Core WiFi, GSM, and NB examples

### PR DESCRIPTION
**Pull request by sandeepmistry**

### CODE REVIEW

**Review by gvarisco — APPROVED** [review]
Awesome work, Sandeep. LGTM.

File: examples/Google Cloud Platform IoT Core/GCP_IoT_Core_GSM/GCP_IoT_Core_GSM.ino [hunk 1]
```diff
@@ -0,0 +1,199 @@
+/*
+  GCP (Google Cloud Platform) IoT Core GSM
+
+  This sketch securely connects to GCP IoT Core using MQTT over GSM/3G.
+  It uses a private key stored in the ATECC508A and a JSON Web Token (JWT) with
+  a JSON Web Signature (JWS).
+
+  It publishes a message every 5 seconds to "/devices/{deviceId}/state" topic
+  and subscribes to messages on the "/devices/{deviceId}/config" and
+  "/devices/{deviceId}/commands/# topics.
```
Comment (per1234) [inline examples/Google Cloud Platform IoT Core/GCP_IoT_Core_GSM/GCP_IoT_Core_GSM.ino]:
```suggestion
  "/devices/{deviceId}/commands/#" topics.
```

File: examples/Google Cloud Platform IoT Core/GCP_IoT_Core_GSM/GCP_IoT_Core_GSM.ino [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    Serial.print(".");
+    delay(1000);
+  }
+
+  Serial.println("You're connected to the cellular network");
+  Serial.println();
+}
+
+void connectMQTT() {
+  Serial.print("Attempting to MQTT broker: ");
```
Comment (per1234) [inline examples/Google Cloud Platform IoT Core/GCP_IoT_Core_GSM/GCP_IoT_Core_GSM.ino]:
```suggestion
  Serial.print("Attempting to connect to MQTT broker: ");
```

File: examples/Google Cloud Platform IoT Core/GCP_IoT_Core_GSM/arduino_secrets.h [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// GSM settings
+#define SECRET_PINNUMBER     ""
+#define SECRET_GPRS_APN      "GPRS_APN" // replace your GPRS APN
```
Comment (per1234) [inline examples/Google Cloud Platform IoT Core/GCP_IoT_Core_GSM/arduino_secrets.h]:
```suggestion
#define SECRET_GPRS_APN      "GPRS_APN" // replace with your GPRS APN
```

File: examples/Google Cloud Platform IoT Core/GCP_IoT_Core_NB/GCP_IoT_Core_NB.ino [hunk 4]
```diff
@@ -0,0 +1,196 @@
+/*
+  GCP (Google Cloud Platform) IoT Core NB
+
+  This sketch securely connects to GCP IoT Core using MQTT over NB IoT/LTE Cat M1.
+  It uses a private key stored in the ATECC508A and a JSON Web Token (JWT) with
+  a JSON Web Signature (JWS).
+
+  It publishes a message every 5 seconds to "/devices/{deviceId}/state" topic
+  and subscribes to messages on the "/devices/{deviceId}/config" and
+  "/devices/{deviceId}/commands/# topics.
```
Comment (per1234) [inline examples/Google Cloud Platform IoT Core/GCP_IoT_Core_NB/GCP_IoT_Core_NB.ino]:
```suggestion
  "/devices/{deviceId}/commands/#" topics.
```

File: examples/Google Cloud Platform IoT Core/GCP_IoT_Core_NB/GCP_IoT_Core_NB.ino [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+    Serial.print(".");
+    delay(1000);
+  }
+
+  Serial.println("You're connected to the cellular network");
+  Serial.println();
+}
+
+void connectMQTT() {
+  Serial.print("Attempting to MQTT broker: ");
```
Comment (per1234) [inline examples/Google Cloud Platform IoT Core/GCP_IoT_Core_NB/GCP_IoT_Core_NB.ino]:
```suggestion
  Serial.print("Attempting to connect to MQTT broker: ");
```

File: examples/Google Cloud Platform IoT Core/GCP_IoT_Core_WiFi/GCP_IoT_Core_WiFi.ino [hunk 6]
```diff
@@ -0,0 +1,193 @@
+/*
+  GCP (Google Cloud Platform) IoT Core WiFi
+
+  This sketch securely connects to GCP IoT Core using MQTT over WiFi.
+  It uses a private key stored in the ATECC508A and a JSON Web Token (JWT) with
+  a JSON Web Signature (JWS).
+
+  It publishes a message every 5 seconds to "/devices/{deviceId}/state" topic
+  and subscribes to messages on the "/devices/{deviceId}/config" and
+  "/devices/{deviceId}/commands/# topics.
```
Comment (per1234) [inline examples/Google Cloud Platform IoT Core/GCP_IoT_Core_WiFi/GCP_IoT_Core_WiFi.ino]:
```suggestion
  "/devices/{deviceId}/commands/#" topics.
```

File: examples/Google Cloud Platform IoT Core/GCP_IoT_Core_WiFi/GCP_IoT_Core_WiFi.ino [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+    delay(5000);
+  }
+  Serial.println();
+
+  Serial.println("You're connected to the network");
+  Serial.println();
+}
+
+void connectMQTT() {
+  Serial.print("Attempting to MQTT broker: ");
```
Comment (per1234) [inline examples/Google Cloud Platform IoT Core/GCP_IoT_Core_WiFi/GCP_IoT_Core_WiFi.ino]:
```suggestion
  Serial.print("Attempting to connect to MQTT broker: ");
```

**Review by per1234 — CHANGES_REQUESTED** [review]

**Comment by Rocketct** [comment]
tested the example, wifi and gsm and works.

**Review by Rocketct — APPROVED** [review]

**Pull request merged**